### PR TITLE
config/task: add check for duplicate config items

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
The user may add duplicate route-rules or filter-rules by mistake which will cause failure while building router or event-filter.
e.g.: route-rules: ["route-rule-1", "route-rule-1"]

### What is changed and how it works?
Add check in `TaskConfig.adjust` and returns error if route-rules/filter-rules are duplicate.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lichunzhu** [comment]
@csuzhangxc Revised, PTAL

File: dm/config/task_test.go [hunk 1]
```diff
@@ -192,4 +202,67 @@ syncers:
...
+	err = ioutil.WriteFile(filepath, configContent, 0644)
+	c.Assert(err, IsNil)
+	taskConfig = NewTaskConfig()
+	err = taskConfig.DecodeFile(filepath)
+	c.Assert(err, IsNil)
+	taskConfig.MySQLInstances[0].RouteRules = []string{"route-rule-1", "route-rule-2", "route-rule-1", "route-rule-2"}
+	taskConfig.MySQLInstances[1].FilterRules = []string{"filter-rule-1", "filter-rule-2", "filter-rule-3", "filter-rule-2"}
+	err = taskConfig.adjust()
+	c.Assert(terror.ErrConfigDuplicateCfgItem.Equal(err), IsTrue)
+	c.Assert(err, ErrorMatches, "[\\s\\S]*mysql-instance\\(0\\)'s route-rules: route-rule-1, route-rule-2[\\s\\S]*")
```
Comment (csuzhangxc) [inline dm/config/task_test.go]: you can use ``` ` ``` instead of ` " `to wrap a literal string, then it can be more readable.

**Comment by csuzhangxc** [comment]
@WangXiangUSTC PTAL

File: dm/config/task.go [hunk 2]
```diff
@@ -516,3 +529,21 @@ func (c *TaskConfig) SubTaskConfigs(sources map[string]DBConfig) ([]*SubTaskConf
...
 }
+
+// checkDuplicateString checks whether the given string array has duplicate string item
+// if there is duplicate, it will return **all** the duplicate strings
+func checkDuplicateString(ruleNames []string) []string {
+	mp := make(map[string]bool, len(ruleNames))
+	dupeArray := make([]string, 0)
+	for _, name := range ruleNames {
+		if added, ok := mp[name]; ok {
+			if !added {
```
Comment (WangXiangUSTC) [inline dm/config/task.go]: seems that don't need judge `added`, if `ok` is true means this `ruleName` is duplicate

Comment (csuzhangxc) [reply dm/config/task.go]: I think its used to prevent `append` more than once.

Comment (lichunzhu) [reply dm/config/task.go]: added is used to avoid to print "duplicate" duplicates.
For example, ["a", "a", "a"], "a" will be appened for two times.

**Pull request merged**